pkg/sql/plan/shuffle.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plan
```

Comment:
why invent your own hash function?     something from standard library or well known, actually we must have our own hash function somewhere.

Reply:
because this is fast and enough for this situation

Reply:
Our own hash function should be much faster than this function.